drivers/hmis/app/models/hmis/hud/enrollment.rb

```diff
@@ -336,6 +336,58 @@ def data_collection_features
     end.compact
   end
 
+  def occurrence_point_forms
+    # All instances for Occurrence Point forms, including inactive/retired, so we can check for legacy data.
+    # Draft forms should never have legacy data, but let's include them just in case.
+    base_scope = Hmis::Form::Instance.with_role(:OCCURRENCE_POINT)
+
+    # All possible form identifiers used for Occurrence Point collection
+    occurrence_point_identifiers = base_scope.pluck(:definition_identifier).uniq
+
+    # Get the latest version of each form definition so we can check for legacy data
+    definitions = Hmis::Form::Definition.where(identifier: occurrence_point_identifiers).latest_versions
```

Comment:
This feels a bit roundabout- the `role` column is actually on the _definitions_ table, not the instances table (`with_role` does a join) so we are doing an unnecessary join instead of just saying `definition.where(role: : OCCURRENCE_POINT)`.

Can we simplify to something like this?


```ruby
definitions = Hmis::Form::Definition.where(role: :OCCURRENCE_POINT).published_or_retired.latest_versions

definitions.each do |definition|
    best_instance = definition.instances.active.detect_best_instance_for_enrollment(enrollment: self)
end
```